lib/auth/auth_with_roles.go

```diff
@@ -1613,6 +1613,15 @@ func (a *ServerWithRoles) listResourcesWithSort(ctx context.Context, req proto.L
 		}
 
 		servers := types.AppServers(appservers)
+
+		// Removes apps that aren't supported by the UI.
+		// Doesn't remove them if the request is coming from someone listing Apps
+		// to create a resource access request, that way they can still request Apps
+		// via the UI even if they can't be launched from the UI.
+		if !req.GetUseSearchAsRoles() {
+			servers = removeUnsupportedApps(servers)
+		}
+
```

Comment:
this limits how `listResourcesWithSort` works, b/c what if we wanted all the apps? we should create a app specific filter to pass in with `ListResources` request, and then add a test testing this new filter